lib/livebook_web/live/session_live/app_docker_component.ex

```diff
@@ -17,21 +17,41 @@ defmodule LivebookWeb.SessionLive.AppDockerComponent do
     socket = assign(socket, assigns)
     deployment_groups = Provider.deployment_groups(assigns.hub)
 
+    deployment_group =
```

Comment:
Instead of changing the docker component, I would create a new app_teams_component. I think they are going to grow very different overtime and there are practical differences too. For example, we should be able to deploy a notebook with Teams without saving it to disk.

Reply:
I think Jose means deploying without the user explicitly selecting a file to save. We can always dump the notebook to a temporary file when creating the zip.

We could definitely do everything in memory, but using tmpdir and deleting afterwards is less pressure on memory. Though we end up encrypting/decrypting the whole zip in memory at once anyway, but when possible probably better to use disk.

Reply:
In memory is great!

Reply:
I would be more worried about decrypting and unzipping, because that's on agent instances where memory may be more limited. The upper limit for attachment we have is 20mb, so if someone has *many* of them it may be an issue, but then it's not really a scenario I would expect either. But yeah, we can revisit if necessary.

@aleDsz so for the notebook we can export to livemd in memory and for attachments, instead of `FileSystem.File.copy` we can do `FileSystem.File.read`. Just make sure the attachments are in `files/` subdir in the zip.

Reply:
Ah, the 20MB is for the whole zip, not each attachment, so it's ok.

@aleDsz we also should validate the size limit here, before attempting the deployment. So we can show the error right away.

Reply:
Good catch! I'll add mostly the same validation as we have in the Teams side